lib/PHPCfg/AstVisitor/NameResolver.php

```diff
@@ -81,7 +84,7 @@ protected function parseTypeDecl($type) {
         }
         $regex = '(^([a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*\\\\)*[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*$)';
         if (!preg_match($regex, $type)) {
-            throw new \RuntimeException("Unknown type declaration found: $type");
+            return false;
```

Comment:
Yeah, those cases can be merged. According to https://github.com/phpDocumentor/fig-standards/blob/master/proposed/phpdoc.md#appendix-a-types `$this` is a valid type (for returns), but in any case the handling would still be the same (i.e. return as-is).